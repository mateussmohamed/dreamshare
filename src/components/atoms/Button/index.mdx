---
name: Button
menu: Molecules
---

import { Playground, PropsTable } from 'docz'
import Button from './index'

# Button
<PropsTable of={Button} />


<Playground>
  <div style={{ background: 'grey', padding: 40 }}>
    <Button>
      Default
    </Button>

    <Button primary>
      Primary
    </Button>

    <Button secondary>
      Secondary
    </Button>

    <Button ghost>
      Ghost
    </Button>
  </div>
</Playground>
