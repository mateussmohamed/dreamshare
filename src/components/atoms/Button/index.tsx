import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'

const defaultStyle = css`
  border: 2px solid #fff;
  background-color: rgba(255, 255, 255, 0);
  color: #fff;

  :hover {
    background-color: #fff;
    border-color: #fff;
    color: ${props => props.theme.colors.brownGrey};
  }
`

const primaryStyle = css`
  border: 2px solid ${props => props.theme.colors.lightRed};
  background-color: rgba(255, 255, 255, 0);
  color: ${props => props.theme.colors.lightRed};
  font-weight: bold;

  :hover {
    background-color: ${props => props.theme.colors.lightRed};
    border-color: ${props => props.theme.colors.lightRed};
    color: #fff;
  }
`

const secondaryStyle = css`
  border: 0;
  background-color: ${props => props.theme.colors.lightRed};
  color: #fff;
  font-weight: bold;

  :hover {
    background-color: #c7191b;
    color: #fff;
  }
`

const ghostStyle = css`
  color: #fff;
  border: 0;
  background-color: rgba(255, 255, 255, 0);

  :hover {
    color: ${props => props.theme.colors.brownGrey};
    border: 2px solid #fff;
  }
`

interface ButtonProps {
  readonly children: React.ReactNode
  readonly primary: boolean
  readonly secondary: boolean
  readonly ghost: boolean
}

const ButtonStyled = styled.button<ButtonProps>`
  ${defaultStyle};
  ${props => props.theme.typography.montserratStyle};
  ${props => props.primary && primaryStyle};
  ${props => props.secondary && secondaryStyle};
  ${props => props.ghost && ghostStyle};
  border-radius: 40px;
  margin: 0.25em;
  padding: 0.25em 1em;
  min-width: 90px;
  height: 40px;
  outline: 0;
  cursor: pointer;
  font-size: 14px;
  transition: color 0.2s ease-in-out, background-color 0.2s ease-in-out,
    border 0.2s ease-in-out;
`

export const Button = props => (
  <ButtonStyled {...props}>{props.children}</ButtonStyled>
)

Button.propTypes = {
  children: PropTypes.node.isRequired,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  ghost: PropTypes.bool,
}
