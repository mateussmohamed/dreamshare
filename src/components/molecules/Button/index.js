import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import {
  border,
  borderRadius,
  transparentize,
  margin,
  padding,
  transitions,
  em,
} from 'polished'

const defaultStyle = css`
  ${border('2px', 'solid', '#fff')};
  background-color: ${transparentize(100, '#fff')};
  color: #fff;

  :hover {
    background-color: #fff;
    border-color: #fff;
    color: ${props => props.theme.colors.brownGrey};
  }
`

const primaryStyle = css`
  ${border('2px', 'solid', '#ff4e50')};
  background-color: ${transparentize(100, '#fff')};
  color: #ff4e50;
  font-weight: bold;

  :hover {
    background-color: #ff4e50;
    border-color: #ff4e50;
    color: #fff;
  }
`

const secondaryStyle = css`
  border: 0;
  background-color: #ff4e50;
  color: #fff;
  font-weight: bold;

  :hover {
    background-color: #ff4e50;
    color: #fff;
  }
`

const ghostStyle = css`
  color: #fff;
  border: 0;
  background-color: ${transparentize(100, '#fff')};

  :hover {
    color: ${props => props.theme.colors.brownGrey};
    ${border('2px', 'solid', '#fff')};
  }
`

const ButtonStyled = styled.button`
  ${borderRadius('top', '40px')};
  ${borderRadius('bottom', '40px')};
  ${margin('0.25em')};
  ${padding('0.25em', '1em', '0.25em', '1em')};
  ${transitions(['color', 'background-color', 'border'], '.2s ease-in-out')};

  display: 'inline-flex';
  min-width: 90px;
  height: 40px;

  outline: 0;
  cursor: pointer;

  ${defaultStyle};
  ${props => props.theme.typography.montserratStyle};
  font-size: 14px;
  ${props => props.primary && primaryStyle};
  ${props => props.secondary && secondaryStyle};
  ${props => props.ghost && ghostStyle};
`

const Button = props => <ButtonStyled {...props}>{props.children}</ButtonStyled>

Button.propTypes = {
  children: PropTypes.node.isRequired,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  ghost: PropTypes.bool,
}

export default Button
