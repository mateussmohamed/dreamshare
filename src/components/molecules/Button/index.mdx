---
name: Button
menu: Molecules
---

import { Playground, PropsTable } from 'docz'
import Button from './index'
import LayoutTemplate from '../../templates/LayoutTemplate'

# Button
<PropsTable of={Button} />

## Basic usage
<Playground>
  <LayoutTemplate>
    <div style={{ background: '#bfbfbf' }}>
      <Button>
        Default
      </Button>

      <Button primary>
        Primary
      </Button>

      <Button secondary>
        Secondary
      </Button>

      <Button ghost>
        Ghost
      </Button>
    </div>
  </LayoutTemplate>
</Playground>
