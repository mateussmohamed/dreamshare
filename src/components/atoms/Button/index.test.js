import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Button from './index'

describe('[<Button />]', () => {
  const wrap = (props = {}) => shallow(<Button {...props} />).dive()

  test('matches to snapshot', () => {
    const wrapper = wrap({ children: 'I am a Title' })

    expect(wrapper).toMatchSnapshot()
  })

  describe('when pass props', () => {
    test('primary', () => {
      const wrapper = wrap({ children: 'Primay Button', primary: true })

      expect(wrapper).toMatchSnapshot()
    })

    test('secondary', () => {
      const wrapper = wrap({ children: 'Secondary Button', secondary: true })

      expect(wrapper).toMatchSnapshot()
    })

    test('ghost', () => {
      const wrapper = wrap({ children: 'Ghot Button', ghost: true })

      expect(wrapper).toMatchSnapshot()
    })
  })
})
