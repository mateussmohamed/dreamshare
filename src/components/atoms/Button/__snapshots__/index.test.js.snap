// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Button />] matches to snapshot 1`] = `
<Button__ButtonStyled>
  I am a Title
</Button__ButtonStyled>
`;

exports[`[<Button />] when pass props ghost 1`] = `
<Button__ButtonStyled
  ghost={true}
>
  Ghot Button
</Button__ButtonStyled>
`;

exports[`[<Button />] when pass props primary 1`] = `
<Button__ButtonStyled
  primary={true}
>
  Primay Button
</Button__ButtonStyled>
`;

exports[`[<Button />] when pass props secondary 1`] = `
<Button__ButtonStyled
  secondary={true}
>
  Secondary Button
</Button__ButtonStyled>
`;
